### PR TITLE
Changed RadioButtonGroup to add defaultValue

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed RadioButtonGroup to add defaultValue.
Aligns with TextInput, TextArea, Select, DateInput, etc.

#### Where should the reviewer start?

RadioButtonGroup.js

#### What testing has been done on this PR?

Added a unit test

#### How should this be manually tested?

manually

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
Looks good, tested with storybook and functionality is working as expected. Shouldn't `defaultValue` be added to the RadioButtonGroup docs though?

**Comment by ericsoderberghp** [comment]
> Looks good, tested with storybook and functionality is working as expected. Shouldn't `defaultValue` be added to the RadioButtonGroup docs though?

We haven't been adding all standard and straightforward DOM attributes where we don't do anything notable to our documentation. For instance, we describe `defaultValue` for Select and DateInput because we have something unique to say but for TextInput and TextArea we don't since they aren't special in any way.

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ const RadioButtonGroup = forwardRef(
   (
     {
       children,
+      defaultValue,
```
Comment (ShimiSun) [inline src/js/components/RadioButtonGroup/RadioButtonGroup.js:20]: I think we need to add `defaultValue` to doc.js.

Comment (ericsoderberghp) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js:20]: Should we add it to TextInput and TextArea too?

Comment (ShimiSun) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js:20]: I think we should, otherwise, how would users be aware of this optional setting?

Comment (ericsoderberghp) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js:20]: It's a standard DOM attribute. So, just like you can put an `id` on any component, we don't document it.

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**